CMakeLists.txt

```diff
@@ -65,8 +66,15 @@ add_subdirectory(cudamapper)
 add_subdirectory(cudaaligner)
 
 # Add documentation generation.
-set_doxygen_mainpage(${CMAKE_CURRENT_SOURCE_DIR}/README.md)
-add_docs_target("ClaraGenomicsAnalysis" "${CGA_VERSION}")
+if (cga_generate_docs STREQUAL "ON")
```

Comment:
Since ON/OFF doubles up as a Boolean in cmake, we can directly test for the `if(cga_generate_docs)` and in the else case print the message

Reply:
As we said, I created an argument validation method under cmake/Utils.cmake